### PR TITLE
Update flutter_tools for the "unoptimized" engine build flag and the …

### PR DESCRIPTION
**Pull request by jason-simmons**
…new output directory naming scheme

### CODE REVIEW

**Comment by jason-simmons** [comment]
@devoncarew

File: packages/flutter_tools/test/toolchain_test.dart [hunk 1]
```diff
@@ -29,10 +29,6 @@ void main() {
       );
 
       expect(
-        toolConfig.getToolsDirectory(platform: HostPlatform.linux_x64).path,
```
Comment (devoncarew) [inline packages/flutter_tools/test/toolchain_test.dart]: Is it necessary to delete this expectation?

**Comment by devoncarew** [comment]
lgtm

File: packages/flutter_tools/test/toolchain_test.dart (hunk 1 above)
Comment (jason-simmons) [inline packages/flutter_tools/test/toolchain_test.dart]: getToolsDirectory() was removed in this PR

Comment (devoncarew) [inline packages/flutter_tools/test/toolchain_test.dart]: Ah, ok. Looks like it was only in use by tests. At least it was well tested :)

**Pull request merged**